Maryam-Marjoori/homework_3/4-fizzbuzz-homework/app.js

```diff
@@ -0,0 +1,76 @@
+
+// Relying on your newfound knowledge of loops, combine loops and if/else statements together and incrementally build the common fizzbuzz loop.
+
+// - In the loop, every time a number is divisible by **3**, instead of logging the number itself, the word "fizz" should appear.
+// - If the number is divisible by  **5**, the word "buzz" should be logged.
+// - If the number is divisible by both **3** and  **5**, then the word "fizzbuzz" should be logged.
+
+// Follow the steps below.
+
+// ##### Step 1:
+// Construct a for loop that iterates through, and `console.log()`'s out, numbers 1 - 100.
+
+
+// ##### Step 2:
+// Add an `if/else` statement that logs the string `"fizz"` if the value being iterated over is divisible by `3`; otherwise, log out the value.
+
+
+
+// ##### Step 3:
+// Add an `else if` clause that logs the string `"buzz"` if the value being iterated over is divisible by `5`.
+
+
+
+// ##### Step 4:
+// Add an additional `else if` clause that logs the string `"fizzbuzz"` if the value being iterated over is divisible by both `3` and `5`.
+
+
+// ************************************************************ First Approach ************************************************************
+
+for ( var totalNum = 100; totalNum > 0 ; totalNum-- ){
+    if ( (totalNum % 3) === 0 ){
+        console.log("fizz");
+    }
+    if ( (totalNum % 5) === 0 ){
+        console.log("buzz");
+    }
+    if (totalNum % 5 === 0 && totalNum % 3 === 0){
+        console.log("Fizz Buzz");
+    }
+    console.log(totalNum);
+}
+
+// ************************************************************ Second Approach ************************************************************
+/*
+for ( var totalNum = 100; totalNum > 0 ; totalNum-- ){
+    var myOutcome = " ";
+    if (totalNum % 3 === 0){ 
+        console.log(myOutcome += "Fizz");
+    }
+    if (totalNum % 5 === 0){
+        console.log(myOutcome += "Buzz");
+    }
+    else {
+        console.log(totalNum);
+    }
+}
+*/
+
+// ************************************************************ Third Approach ************************************************************
+
+/*
+for (var totalNum = 100; totalNum > 0 ; totalNum-- ) { 
+    console.log((totalNum % 3 ? '':'Fizz')+(totalNum % 5 ?'':'Buzz') || totalNum );
+}
+*/
+
+// ************************************************************ Forth Approach ************************************************************
+
+/*
+for ( var totalNum = 100; totalNum > 0 ; totalNum-- ) {
+  var myThree = totalNum % 3 == 0, 
+        myFive = totalNum % 5 == 0;
+  console.log(myThree ? myFive ? "FizzBuzz" : "Fizz" : myFive ? "Buzz" : totalNum);
```

Comment:
These loops got progressively more exciting. :P 
If you feel comfortable using ternary expressions, I encourage you to keep using them for short conditionals.